src/StreamJsonRpc.Tests/MessagePackFormatter.cs

```diff
@@ -119,5 +171,95 @@ public void Serialize(IBufferWriter<byte> contentBuffer, JsonRpcMessage message)
 
             return result;
         }
+
+        private class StandardPlusIProgressOfTResolver : IFormatterResolver
+        {
+            private readonly MessagePackFormatter formatter;
+
+            public StandardPlusIProgressOfTResolver(MessagePackFormatter formatter)
+            {
+                this.formatter = formatter;
+            }
+
+            public IMessagePackFormatter<T> GetFormatter<T>()
+            {
+                return new FormatterCache<T>(this.formatter).Formatter;
+            }
+
+            private class FormatterCache<T>
+            {
+                public readonly IMessagePackFormatter<T> Formatter;
+
+                public FormatterCache(MessagePackFormatter formatter)
+                {
+                    if (MessageFormatterHelper.FindIProgressOfT(typeof(T)) != null)
+                    {
+                        // Call Get Formatter from IProgress Formatter?
```

Comment:
>// Call Get Formatter from IProgress Formatter? [](start = 24, length = 47)

Your `T` here may not be `IProgress<T>`, but at least on the client side it will be the concrete type that implements that interface. Go ahead and use a `IProgressOfTFormatter<T>` instance of your formatter, which must simply understand that the generic type argument is the concrete type on the client side.
You'll want to reuse that instance of the formatter every time you see that `T` again. I would use a `Dictionary<Type, object>` field (with locking) to cache these as you create them.